android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -2724,6 +2724,32 @@ private void setAmazonResult(final Object result, final String adUnitId, final M
         }
     }
 
+    public void setAmazonResult(final Object result, final Object adView)
```

Comment:
We should allow pub to call into plugin via `public void setAmazonBannerResult(final Object result, final String adUnitId)`, etc. 

Take `adUnitId` from there, and get adview from `android/src/main/java/com/applovin/reactnative/AppLovinMAXAdViewManager.java`. If native UI component adview exists call into that, otherwise default to `MaxAdView adView = retrieveAdView( adUnitId, adFormat );`.

Reply:
But we should always try to compact similar logic instead of creating a new method that is very similar to what we arleady have.

Reply:
Ok, I have now added the iOS version.